cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/AbstractContext.java

```diff
@@ -169,7 +169,7 @@ public DiscoveryServiceClient getDiscoveryServiceClient() {
     return discoveryServiceClient;
   }
 
-  protected static Map<String, String> getMetricsContext(Program program, String runId) {
+  public static Map<String, String> getMetricsContext(Program program, String runId) {
```

Comment:
it may be better to extract this out into a MetricsContext class.  Then it is more extensible in the future if we need anything besides tags.